model/host/db.go

```diff
@@ -1238,22 +1238,85 @@ var (
 	awsSecretKey = bsonutil.MustHaveTag(EC2ProviderSettings{}, "Secret")
 )
 
-func StartingHostsByClient(ctx context.Context, limit int) (map[ClientOptions][]Host, error) {
+// FindStartingHostsByClient returns a map of cloud provider client options to hosts
+// with those client options that are starting up. The limit limits the number
+// of hosts that can be returned. The limit is applied separately for task hosts
+// and spawn hosts/host.create hosts.
+func FindStartingHostsByClient(ctx context.Context, limit int) ([]HostsByClient, error) {
 	if limit <= 0 {
 		limit = 500
 	}
-	results := []struct {
-		Options ClientOptions `bson:"_id"`
-		Hosts   []Host        `bson:"hosts"`
-	}{}
 
-	pipeline := []bson.M{
+	nonTaskHosts, err := findStartingNonTaskHosts(ctx, limit)
+	if err != nil {
+		return nil, errors.Wrap(err, "finding starting non-task hosts")
+	}
+
+	taskHostsByClient, err := findStartingTaskHosts(ctx, limit)
+	if err != nil {
+		return nil, errors.Wrap(err, "finding starting task hosts")
+	}
+
+	hostsByClient := []HostsByClient{}
```

Comment:
I think these 3 lines can be replaced with `hostsByClient := append(nonTaskHosts, taskHostsByClient...)`